### PR TITLE
Update algorithms methods to match proposal

### PR DESCRIPTION
**Pull request by natecook1000**
- [x] Sequence/Collection constraints
- [x] `@escaping`/`rethrows` closures
- [x] `split` parameters
- [x] More exhaustive testing
- [ ] ~Documentation comments~ _Coming in a separate PR_

### CODE REVIEW

**Comment by natecook1000** [comment]
@swift-ci Please test

**Comment by natecook1000** [comment]
@swift-ci Please test

**Comment by natecook1000** [comment]
@swift-ci Please test

File: Sources/_StringProcessing/Algorithms/Algorithms/Contains.swift [hunk 1]
```diff
@@ -46,6 +46,20 @@ extension BidirectionalCollection where Element: Comparable {
...
+  @available(SwiftStdlib 5.7, *)
+  public func contains(_ other: String) -> Bool {
+    firstRange(of: other) != nil
+  }
+  
+  @available(SwiftStdlib 5.7, *)
+  public func contains(_ other: Substring) -> Bool {
+    firstRange(of: other) != nil
+  }
+}
```
Comment (itingliu) [inline Sources/_StringProcessing/Algorithms/Algorithms/Contains.swift:61]: Do we need to add these to the proposal?

**Review by itingliu — APPROVED** [review]
Looking great!

File: Sources/_StringProcessing/Algorithms/Algorithms/Contains.swift (hunk 1 above)
Comment (natecook1000) [reply Sources/_StringProcessing/Algorithms/Algorithms/Contains.swift:61]: I don't think so - we just need to make sure these are winning the overload resolution vs Foundation's overlay version. However we do that is just an implementation detail.

**Pull request merged**